Assets/ZSSRichTextEditor.js

```diff
@@ -806,8 +806,446 @@ ZSSEditor.removeImage = function(imageNodeIdentifier) {
+        height: '',         // The image height attribute
+        linkClassName: '',  // The class attribute for the link
+        linkRel: '',        // The rel attribute for the link (if any)
+        linkTargetBlank: false, // true if the link should open in a new window.
+        linkUrl: '',        // The href attribute of the link
+        size: 'custom',     // Accepted values: custom, medium, large, thumbnail, or empty string
+        src: '',            // The src attribute of the image
+        title: '',          // The title attribute of the image (if any)
+        width: ''           // The image width attribute
+    };
+
+    // populate metadata with values of matched attributes
+    metadata.src = $( imageNode ).attr( 'src' ) || '';
+    metadata.alt = $( imageNode ).attr( 'alt' ) || '';
+    metadata.title = $( imageNode ).attr( 'title' ) || '';
+
+    width = $(imageNode).attr( 'width' );
+    height = $(imageNode).attr( 'height' );
+
+    if ( ! isIntRegExp.test( width ) || parseInt( width, 10 ) < 1 ) {
+        width = imageNode.naturalWidth || imageNode.width;
+    }
+
+    if ( ! isIntRegExp.test( height ) || parseInt( height, 10 ) < 1 ) {
+        height = imageNode.naturalHeight || imageNode.height;
+    }
+
+    metadata.width = width;
+    metadata.height = height;
+
+    classes = imageNode.className.split( /\s+/ );
+    extraClasses = [];
+
+    $.each( classes, function( index, value ) {
+        if ( /^wp-image/.test( value ) ) {
+           metadata.attachment_id = parseInt( value.replace( 'wp-image-', '' ), 10 );
+        } else if ( /^align/.test( value ) ) {
+           metadata.align = value.replace( 'align', '' );
+        } else if ( /^size/.test( value ) ) {
+           metadata.size = value.replace( 'size-', '' );
+        } else {
+           extraClasses.push( value );
+        }
+    } );
+
+    metadata.classes = extraClasses.join( ' ' );
+
+    // Extract caption
+    var captionMeta = ZSSEditor.captionMetaForImage( imageNode )
+    metadata = $.extend( metadata, captionMeta );
+
+    // Extract linkTo
+    if ( imageNode.parentNode && imageNode.parentNode.nodeName === 'A' ) {
+        link = imageNode.parentNode;
+        metadata.linkClassName = link.className;
+        metadata.linkRel = $( link ).attr( 'rel' ) || '';
+        metadata.linkTargetBlank = $( link ).attr( 'target' ) === '_blank' ? true : false;
+        metadata.linkUrl = $( link ).attr( 'href' ) || '';
+    }
+
+    return metadata;
+};
+
+/**
+ *  @brief      Extracts the caption shortcode for an image.
+ *
+ *  @param      imageNode   An image node in the DOM to inspect.
+ *
+ *  @return     Returns a shortcode match (if any) for the passed image node.
+ *  See shortcode.js::next for details
+ */
+ZSSEditor.getCaptionForImage = function( imageNode ) {
+    var node = imageNode;
+
+    // If the image's parent is an anchor, let the anchor be the working node.
+    if ( node.parentNode && node.parentNode.nodeName === 'A' ) {
```

Comment:
Since this code is almost identical to the code written in `ZSSEditor.selectImageCaptionNode` maybe we can have a method to obtain the caption node (ie: `getCaptionNodeForImage` or something similar).